setup.py

```diff
@@ -93,48 +94,136 @@ def get_scm_config():
+        include_dirs.append(plat_include)
+    if config_h_dir not in (default_include, plat_include):
+        include_dirs.append(config_h_dir)
+    return include_dirs + HPY_INCLUDE_DIRS
+
+
+class build_clib_hpy(build_clib):
+    """ Special build_clib command for building HPy's static libraries defined
+        by 'STATIC_LIBS' below. The behavior differs in following points:
+        (1) Option 'force' is set such that static libs will always be renewed.
+        (2) Method 'get_library_names' always returns 'None'. This is because
+            we only use this command to build static libraries for shipping
+            them. We don't need them for linking here.
+        (3) This command consumes a custom build info key 'abi' that is used
+            to create a separate build temp directories for each ABI and an
+            ABI-specific output directory. This is necessary to avoid incorrect
+            sharing of (temporary) build artifacts.
+        (4) The resulting static library is written to build_ext's output
+            directory. Currently, we use hard-coded sub-directory 'hpy/devel'.
+    """
+    def finalize_options(self):
+        super().finalize_options()
+        self.force = 1
+
+    def get_library_names(self):
+        # We only build static libraries for shipping. We don't want that our
+        # extensions (i.e. 'hpy.universal' etc) link to these libs.
+        return None
+
+    def build_libraries(self, libraries):
+        build = self.get_finalized_command('build')
+        # we just inherit the 'inplace' option from 'build_ext'
+        inplace = self.get_finalized_command('build_ext').inplace
+        if inplace:
+            # the inplace option requires to find the package directory
+            # using the build_py command for that
+            build_py = self.get_finalized_command('build_py')
+            lib_dir = os.path.abspath(build_py.get_package_dir('hpy.devel'))
+        else:
+            lib_dir = os.path.join(build.build_lib, 'hpy', 'devel')
+
+        for lib in libraries:
+            lib_name, build_info = lib
+            abi = build_info.get('abi')
+            # Call super's build_libraries with just one library in the list
+            # such that we can temporarily change the 'build_temp'.
+            orig_build_temp = self.build_temp
+            orig_build_clib = self.build_clib
+            self.build_temp = os.path.join(orig_build_temp, 'lib', abi)
+            self.build_clib = os.path.join(lib_dir, 'lib', abi)
+            try:
+                super().build_libraries([lib])
+            finally:
+                self.build_temp = orig_build_temp
+                self.build_clib = orig_build_clib
+
+
+STATIC_LIBS = [(HPY_EXTRA_LIB_NAME,
+                {'sources': HPY_EXTRA_SOURCES,
+                 'include_dirs': get_hpy_runtime_includes(),
+                 'abi': 'universal',
+                 'macros': [('HPY_UNIVERSAL_ABI', None)]}),
```

Comment:
if I understand correctly `'abi'` is not a distutils thing but it's some custom setting which is used only by us, right?
Then I propose to call it `hpy_abi` (which is also consistent to how it is called everywhere else in the code)

Reply:
Yes, `abi` is not a distutils thing. Okay, I'll rename.